lib/infra/infra-stack.ts

```diff
@@ -412,30 +415,45 @@ export class InfraStack extends Stack {
       this.opensearchPortMapping = parseInt(opensearchPortMap, 10);
     }
 
-    const opensearchListener = nlb.addListener('opensearch', {
-      port: this.opensearchPortMapping,
-      protocol: Protocol.TCP,
-    });
-    if (!this.securityDisabled && !this.minDistribution) {
-      if (certificateArn !== 'undefined') {
-        opensearchListener.addCertificates('cert', [ListenerCertificate.fromArn(certificateArn)]);
-      }
-    }
-
-    const opensearchDashboardsPortMap = `${props?.mapOpensearchDashboardsPortTo ?? scope.node.tryGetContext('mapOpensearchDashboardsPortTo')}`;
     if (opensearchDashboardsPortMap === 'undefined') {
       this.opensearchDashboardsPortMapping = 8443;
     } else {
       this.opensearchDashboardsPortMapping = parseInt(opensearchDashboardsPortMap, 10);
     }
 
-    if (this.dashboardsUrl !== 'undefined') {
-      dashboardsListener = nlb.addListener('dashboards', {
-        port: this.opensearchDashboardsPortMapping,
+    if (this.opensearchPortMapping === this.opensearchDashboardsPortMapping) {
+      throw new Error('OpenSearch and OpenSearch-Dashboards cannot be mapped to the same port! Please provide different port numbers.'
+      + ` Current mapping is OpenSearch:${this.opensearchPortMapping} OpenSearch-Dashboards:${this.opensearchDashboardsPortMapping}`);
+    }
+
+    if (!this.securityDisabled && !this.minDistribution && this.opensearchPortMapping === 443 && certificateArn !== 'undefined') {
```

Comment:
So it is possible for either os or osd to bind to 443, but not both?

In that case what if someone wants to get TLS on both os and osd?

Reply:
For now we have enforced that TLS can only be enabled for 443 port. It can be any OS or OSD and certificate ARN is compulsory.

Reply:
For now we decided to go with HTTPS port with TLS. Can surely be extended in future for all ports.